src/compilersupport_p.h

```diff
@@ -44,7 +44,7 @@
 #  include <stdbool.h>
 #endif
 
-#if __STDC_VERSION__ >= 201112L || __cplusplus >= 201103L || __cpp_static_assert >= 200410
+#if __STDC_VERSION__ >= 201112L || (defined(__cplusplus) && __cplusplus >= 201103L) || __cpp_static_assert >= 200410
```

Comment:
You'll likely need a similar change to the 3rd component here, __cpp_static_assert is pretty old, but still should probably be guarded in the same way (particularly since it wouldn't be included if someone compiles with -std=c++98).

Reply:
done!